From af54068a01a5c9e32bb2fdef3d2fb6191eb2f0a5 Mon Sep 17 00:00:00 2001
From: Venkateshwarlu Domakonda <Venkateshwarlu@codeaurora.org>
Date: Thu, 22 Nov 2012 19:35:13 +0530
Subject: [PATCH] qcom-fm: Add proper permission for FM to get/set property
 variables

Allow FM Process to set and get the system property variables
specific for FM operation.

Change-Id: Idc937e5c3ca4fbb9fbfe2e4e9eed9eb9bd50499c

Conflicts:
	include/private/android_filesystem_config.h
---
 include/private/android_filesystem_config.h | 6 ++++++
 init/property_service.c                     | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/include/private/android_filesystem_config.h b/include/private/android_filesystem_config.h
index 16efe58..73f35c4 100644
--- a/include/private/android_filesystem_config.h
+++ b/include/private/android_filesystem_config.h
@@ -62,6 +62,9 @@
 #define AID_DRMRPC        1026  /* group for drm rpc */
 #define AID_NFC           1027  /* nfc subsystem */
 #define AID_SDCARD_R      1028  /* external storage read access */
+#ifdef QCOM_FM_ENABLED
+#define AID_FM_RADIO      1029  /* FM radio */
+#endif
 #define AID_AUDIT         1031  /* audit daemon */
 
 #define AID_THEMEMAN      1300  /* theme manager */
@@ -176,6 +179,9 @@ static const struct android_id_info android_ids[] = {
     { "mot_dlna",  AID_MOT_DLNA, },
 #endif
     { "misc",      AID_MISC, },
+#ifdef QCOM_FM_ENABLED
+    { "fm_radio",  AID_FM_RADIO, },
+#endif
     { "nobody",    AID_NOBODY, },
     { "theme_man", AID_THEMEMAN },
     { "audit",      AID_AUDIT, },
diff --git a/init/property_service.c b/init/property_service.c
index d3558e7..6e8cb5d 100755
--- a/init/property_service.c
+++ b/init/property_service.c
@@ -100,6 +100,9 @@ struct {
     { "persist.security.", AID_SYSTEM,   0 },
     { "persist.service.bdroid.", AID_BLUETOOTH,   0 },
     { "selinux."         , AID_SYSTEM,   0 },
+#ifdef QCOM_FM_ENABLED
+    { "hw.fm.",           AID_FM_RADIO,  0 },
+#endif
     { "net.pdp",          AID_RADIO,    AID_RADIO },
     { "service.bootanim.exit", AID_GRAPHICS, 0 },
 #ifdef PROPERTY_PERMS_APPEND
@@ -122,6 +125,9 @@ struct {
 } control_perms[] = {
     { "dumpstate",AID_SHELL, AID_LOG },
     { "ril-daemon",AID_RADIO, AID_RADIO },
+#ifdef QCOM_FM_ENABLED
+    { "fm_dl", AID_FM_RADIO, AID_FM_RADIO},
+#endif
 #ifdef CONTROL_PERMS_APPEND
 CONTROL_PERMS_APPEND
 #endif
-- 
1.8.4.msysgit.0

